popup.html

```diff
@@ -7,7 +7,9 @@
 </head>
 <body>
     <button id="toggleRedirect" type="button"></button>
-    <script src="browser-wrapper.js"></script>
-    <script src="popup.js"></script>
+    <br><br>
+    Python Version: <input type="text" id="pythonVersion" size="3" maxlength="3" />
```

Comment:
This is how it looks like on my Firefox:
![Screenshot from 2020-05-05 21-30-48](https://user-images.githubusercontent.com/3128220/81107810-08550c80-8f18-11ea-8dbe-7ea59793563e.png)
Because of that, I would recommend putting the input in a separate line than the "Python version" label, as well as setting font size and possibly font family explicitly to avoid differences between platforms / web browsers. Optionally, setting all of the input appearance (border, outline, font, etc.) could help too.

Reply:
Ah, got it. I'll stop being lazy and just download firefox so I can test and fix this :-)

Reply:
Ok, this is what it looks like in FF on my machine (MacOS)

![Screen Shot 2020-05-05 at 9 22 57 PM](https://user-images.githubusercontent.com/214370/81130715-bcee2e80-8f16-11ea-8350-870027f3903c.png)

Styling is based on fonts from docs.python.org

Reply:
I think you should hard code `3.` and make the second number a drop down going from 0 or 5 to 9.

Reply:
I played around with having a dropdown - what I really wanted to do was get the dropdown from the page itself, but that got more complicated since it's generated by javascript.

If we do a dropdown with fixed version numbers that means the plugin will need to be updated when 3.10 is available or when one of the previous versions gets its docs removed. If that's ok then happy to make it a select, just let me know.

Reply:
**tldr: I think that the fixed version numbers approach is okay**

Long version:

I can see a few options here. The list of versions could be retrieved from several places:
* https://www.python.org/doc/versions/ - requires parsing HTML, is possible to break in the future (when the HTML changes)
* https://docs.python.org/3.8/_static/switchers.js - from `all_versions` variable. Again, possible to break in the future (when the JS changes). Here, we should parse the JS as well (I'd like to avoid executing external JavaScript, as it generates additional problems during Firefox add-ons review, and is prone to be insecure)
* We could as well iterate through the versions by requesting https://docs.python.org/3.0/, https://docs.python.org/3.1/, etc. until we get 404. Pretty good, since it'll pretty much work forever, although requires *\<number of Python versions\>* requests to be made.

Another problem is: when to check for new versions? I'd say we could cache the results in `localStorage` and possibly check again if there has been, let's say a week since the last check to avoid making too many requests. I believe the check should be made when opening the popup (so when you don't change the Python version, or disable redirecting, those checks are not done at all).

Considering all of that, this will increase the complexity of the PR, so I'm not sure if it's worth it. Because new Python versions come out about once a year (https://en.wikipedia.org/wiki/History_of_Python#Version_3), I think it'll be okay if I just update them manually. :P

> when one of the previous versions gets its docs removed

Unlikely to happen, as you can still access Python 1.4 docs: https://docs.python.org/release/1.4/lib/

Reply:
I was trying to get the version lists directly from the page. Since `popup.html` is only accessible when the extension is active, that means that it's looking at a `docs.python.org` page which means that it can see the version selector.

But I agree that this is way too overcomplicated for a once a year tiny tweak - and probably a bigger risk of breaking if they change the page HTML/JS :-) I'll make the diff to use a static selector and prop it.

Reply:
About the `<input>` — this is how it looks like on my Linux machine now:

![Screenshot from 2020-05-06 18-55-20](https://user-images.githubusercontent.com/3128220/81205826-82929900-8fcb-11ea-908b-db44ef34e4c7.png)

Better, but still not perfect. It's still dependent on the system theme (which means it might be an issue on Windows as well). I think when/if we make this a select instead of input, the `Python version` label could be dropped completely, as it will be pretty obvious then what it is for (we could possibly add `title` attribute to be extra safe).

Reply:
Got it, will tweak it - let's shoot for good, not perfect :-)

Reply:
Ok, set, lmk what you think